projects/gateway2/translator/sslutils/ssl_utils.go

```diff
@@ -59,3 +78,80 @@ func cleanedSslKeyPair(certChain, privateKey, rootCa string) (cleanedChain strin
 
 	return cleanedChain, err
 }
+
+type SslExtensionOptionFunc = func(ctx context.Context, in string, out *ssl.SslConfig) error
+
+func ApplyCipherSuites(ctx context.Context, in string, out *ssl.SslConfig) error {
+	cipherSuites := strings.Split(in, ",")
+	out.Parameters.CipherSuites = cipherSuites
+	return nil
+}
+
+func ApplyMinimumTlsVersion(ctx context.Context, in string, out *ssl.SslConfig) error {
+	if parsed, ok := ssl.SslParameters_ProtocolVersion_value[in]; ok {
+		out.Parameters.MinimumProtocolVersion = ssl.SslParameters_ProtocolVersion(parsed)
+		if out.Parameters.MaximumProtocolVersion != ssl.SslParameters_TLS_AUTO && !(out.Parameters.MinimumProtocolVersion <= out.Parameters.MaximumProtocolVersion) {
+			return eris.Errorf("maximum tls version %s is less than minimum tls version %s", in, out.Parameters.MinimumProtocolVersion.String())
+		}
+	} else {
+		return eris.Errorf("invalid minimum tls version: %s", in)
+	}
+	return nil
+}
+
+func ApplyMaximumTlsVersion(ctx context.Context, in string, out *ssl.SslConfig) error {
+	if parsed, ok := ssl.SslParameters_ProtocolVersion_value[in]; ok {
+		out.Parameters.MaximumProtocolVersion = ssl.SslParameters_ProtocolVersion(parsed)
+		if out.Parameters.MaximumProtocolVersion != ssl.SslParameters_TLS_AUTO && !(out.Parameters.MinimumProtocolVersion <= out.Parameters.MaximumProtocolVersion) {
+			return eris.Errorf("maximum tls version %s is less than minimum tls version %s", in, out.Parameters.MinimumProtocolVersion.String())
+		}
+	} else {
+		contextutils.LoggerFrom(ctx).Debugf("invalid maximum tls version: %s", in)
+	}
+	return nil
+}
+
+func ApplyOneWayTls(ctx context.Context, in string, out *ssl.SslConfig) error {
+	if strings.ToLower(in) == "true" {
+		out.OneWayTls = wrapperspb.Bool(true)
+	}
+	return nil
+}
+
+func ApplyVerifySubjectAltName(ctx context.Context, in string, out *ssl.SslConfig) error {
+	altNames := strings.Split(in, ",")
+	out.VerifySubjectAltName = altNames
+	return nil
+}
+
+var SslExtensionOptionFuncs = map[string]SslExtensionOptionFunc{
+	GatewaySslCipherSuites:         ApplyCipherSuites,
+	GatewaySslMinimumTlsVersion:    ApplyMinimumTlsVersion,
+	GatewaySslMaximumTlsVersion:    ApplyMaximumTlsVersion,
+	GatewaySslOneWayTls:            ApplyOneWayTls,
+	GatewaySslVerifySubjectAltName: ApplyVerifySubjectAltName,
+}
+
+// ApplySslExtensionOptions applies the GatewayTLSConfig options to the SslConfig
+// This function will never exit early, even if an error is encountered.
+// It will apply all options and return all errors encountered.
+func ApplySslExtensionOptions(ctx context.Context, in *gwv1.GatewayTLSConfig, out *ssl.SslConfig) {
+	if out.Parameters == nil {
+		out.Parameters = &ssl.SslParameters{}
+	}
```

Comment:
I don't love having to set to empty in the caller and then the `SslExtensionOptionFunc`s assuming it won't be nil; seems like not ideal separation of concerns? But I also recognize it avoids duplication. Curious about others' thoughts.